client/src/featureform/register.py

```diff
@@ -1163,44 +1164,32 @@ def register_resources(
             timestamp_column=timestamp_column,
             description=description,
             schedule=schedule,
+            client_object=self,
         )
 
     @staticmethod
     def _assert_query_contains_at_least_one_source(query):
         # Checks to verify that the query contains a FROM {{ name.variant }}
-        pattern = r"from\s*\{\{\s*[a-zA-Z0-9_]+(\.[a-zA-Z0-9_]+)?\s*\}\}"
+        pattern = r"from\s*\{\{\s*[a-zA-Z0-9_:.]+(\.[a-zA-Z0-9_:.]+)?\s*\}\}"
```

Comment:
this is so variant in sql transformation can have timestamp variants

Reply:
Are we going to run into issues with the colons on any of the providers?

Reply:
Or in the URL?

Reply:
I think it should be fine for the URL as long as things are encoded. Not sure what to do if people put ':' in their provider names, we don't have any name validation in place yet